File: SPCode/CS/Docking/Extender/FromBase/Helpers/ResourceHelper.cs
using System.Text.RegularExpressions;
using System.Diagnostics;
using System;
using System.Drawing.Drawing2D;
using System.Windows.Forms;
using System.Collections;
using System.Drawing;
using Microsoft.VisualBasic;
using System.Data.SqlClient;
using System.Data;
using System.Collections.Generic;
using WeifenLuo.WinFormsUI;
using Microsoft.Win32;
using WeifenLuo;
using System.Reflection;
using System.Resources;

// *****************************************************************************
//
//  Copyright 2004, Weifen Luo
//  All rights reserved. The software and associated documentation
//  supplied hereunder are the proprietary information of Weifen Luo
//  and are supplied subject to licence terms.
//
//  WinFormsUI Library Version 1.0
// *****************************************************************************
namespace SoftLogik.Win
{
	namespace UI
	{
		namespace Docking
		{
			
			internal class ResourceHelper
			{
				
				private static ResourceManager m_resourceManager;
				static ResourceHelper()
				{
					m_resourceManager = new ResourceManager("Strings", typeof(ResourceHelper).Assembly);
				}
				public static Bitmap LoadBitmap(string name)
				{
					Assembly Assembly = typeof(DockPanel).Assembly;
					string fullNamePrefix = "WeifenLuo.WinFormsUI.Resources.";
					return new Bitmap(Assembly.GetManifestResourceStream(fullNamePrefix + name));
				}
				public static Bitmap LoadExtenderBitmap(string name)
				{
					//On Error Resume Next VBConversions Warning: On Error Resume Next not supported in C#
					Bitmap outBitmap = (Bitmap) (Image.FromStream(Assembly.GetExecutingAssembly().GetManifestResourceStream("SoftLogik.Win." + name)));
					return outBitmap;
				}
				public static string GetString(string name)
				{
					//On Error Resume Next VBConversions Warning: On Error Resume Next not supported in C#
					return m_resourceManager.GetString(name);
				}
			}
		}
	}
	
}

File: SPCode/Docking/Extender/FromBase/Helpers/ResourceHelper.vb
' *****************************************************************************
' 
'  Copyright 2004, Weifen Luo
'  All rights reserved. The software and associated documentation 
'  supplied hereunder are the proprietary information of Weifen Luo
'  and are supplied subject to licence terms.
' 
'  WinFormsUI Library Version 1.0
' *****************************************************************************
Imports System
Imports System.Drawing
Imports System.Reflection
Imports System.Resources
Imports System.Windows.Forms
Imports WeifenLuo.WinFormsUI
Namespace UI.Docking
    Friend Class ResourceHelper
        Private Shared m_resourceManager As ResourceManager
        Shared Sub New()
            m_resourceManager = New ResourceManager("Strings", GetType(ResourceHelper).Assembly)
        End Sub
        Public Shared Function LoadBitmap(ByVal name As String) As Bitmap
            Dim assembly As Assembly = GetType(DockPanel).Assembly
            Dim fullNamePrefix As String = "WeifenLuo.WinFormsUI.Resources."
            Return New Bitmap(assembly.GetManifestResourceStream(fullNamePrefix & name))
        End Function
        Public Shared Function LoadExtenderBitmap(ByVal name As String) As Bitmap
            On Error Resume Next
            Dim outBitmap As Bitmap = CType(Image.FromStream(Assembly.GetExecutingAssembly().GetManifestResourceStream("SoftLogik.Win." & name)), Bitmap)
            Return outBitmap
        End Function
        Public Shared Function GetString(ByVal name As String) As String
            On Error Resume Next
            Return m_resourceManager.GetString(name)
        End Function
    End Class
End Namespace
